### PR TITLE
Added Lambdas and ECS access to environment variables 

### PR DESCRIPTION
**Pull request by MurasatoNaoya**
All compute related to all pipelines for the StarWatch project now have access to environment variables at run time. Providing access to these variables within the container itself is unsafe, as a container's contents can be inspected, potentially exposing sensitive information. 

To avoid this, the necessary environment variables have been baked into the Terraform build itself, meaning they are injected into Lambdas running images from ECR at runtime. As the terraform.tfvars is not pushed to GitHub and remain on local machines, environment variables are kept safe when used in containers.

This closes: #150

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -169,19 +195,16 @@ resource "aws_lambda_function" "lambda_functions" {
...
-		DB_NAME      = var.DB_NAME
-		DB_USER      = var.DB_USER
-		DB_PASSWORD  = var.DB_PASSWORD
+# All Lamda fucntions will have access to these tf.vars env variables at run time.
+# Meaning scripts run in a Lambda can access them by default, like so for example - 
+# 'os.environ['DB_HOST']' in Python. This uses the pre-defined a locals on line 150.
+  	environment {
+    	variables = local.common_env_vars
+		}
 	}
```
Comment (AlexOBarnes) [inline terraform/main.tf:207]: Really like the iteration

File: terraform/main.tf [hunk 2]
```diff
@@ -32,9 +32,35 @@ data "aws_ecs_cluster" "c13-cluster" {
...
+    # API Keys needed for Lambdas running pipelines.
+    NASA_API_KEY = var.NASA_API_KEY
+    ASTRONOMY_KEY_ID  = var.ASTRONOMY_KEY_ID
+	ASTRONOMY_SECRET = var.ASTRONOMY_SECRET
+		
+
+	# More to be added when other pipeline come online..
+  }
+}
+
```
Comment (AlexOBarnes) [inline terraform/main.tf:60]: Really like this function

Comment (olliejth) [reply terraform/main.tf:60]: Agree with the decision to include all environment variables in the terraform.

File: terraform/main.tf [hunk 3]
```diff
@@ -54,7 +80,7 @@ resource "aws_security_group" "c13-andrew-starwatch-rds-sg" {
 
 
 # Security group for the ECS service hosting the dashboard.
-# Ingress is set to any traffic on port 8501, crucial for
+# Ingress is set to any traffic on the specific dashboard port, crucial for
 # people accessing the dashboard online, as in-bound traffic is blocked by default.
```
Comment (olliejth) [inline terraform/main.tf:84]: What's the reason for allowing this?

File: terraform/main.tf [hunk 4]
```diff
@@ -32,9 +32,35 @@ data "aws_ecs_cluster" "c13-cluster" {
   cluster_name = var.CLUSTER_NAME
 }
 
+# This locals defined the environment variables needed by our
+# Lambdas and ECS dashboard service. In the variables.tf file 
+# a sensitive tag is sometimes applied to ensure the values themselves
+# don't appear in terminal logs, or ouputs of any kind. Truly secure!
+locals {
```
Comment (CalebMathers) [inline terraform/main.tf:39]: Great way to declare environment variables

File: terraform/main.tf (hunk 2 above)
Comment (CalebMathers) [reply terraform/main.tf:60]: I agree with the agreement

File: terraform/main.tf [hunk 5]
```diff
@@ -66,7 +92,7 @@ resource "aws_security_group" "c13_ecs_service_sg" {
 		from_port   = 8501
 		to_port     = 8501
 		protocol    = "tcp"
-		cidr_blocks = ["0.0.0.0/0"]  # Allows access from anywhere
+		cidr_blocks = ["0.0.0.0/0"]  # Allows access from anywhere, for ease in this project.
```
Comment (CalebMathers) [inline terraform/main.tf:95]: Nice specific comment

File: terraform/main.tf [hunk 6]
```diff
@@ -496,30 +519,23 @@ resource "aws_ecs_task_definition" "c13_starwatch_task" {
...
-			name  = "DB_USER"
-			value = var.DB_USER
-			},
-			{
-			name  = "DB_PASSWORD"
-			value = var.DB_PASSWORD
+			# The environement the ECS service runs in will have access
+			# to all environment variables in local.common_env_vars.
+			environment = [
+			for key, value in local.common_env_vars : {
```
Comment (CalebMathers) [inline terraform/main.tf:533]: Good iteration, very smart!

**Review by CalebMathers — APPROVED** [review]

File: terraform/variables.tf [hunk 7]
```diff
@@ -74,3 +81,24 @@ variable "DASHBOARD_IMAGE_URI" {
...
+variable "ASTRONOMY_KEY_ID" {
+  type = string
+  sensitive   = true
+  
+}
+
+variable "ASTRONOMY_SECRET" {
+  type = string
+  sensitive   = true
+
```
Comment (olliejth) [inline terraform/variables.tf:103]: Consistent formatting of variables.

**Pull request merged**